src/async_fuse/memfs/kv_engine.rs

```diff
@@ -67,6 +78,30 @@ impl ValueType {
             _ => panic!("expect ValueType::INum but get {self:?}"),
         }
     }
+
+    /// Turn the `ValueType` into `String`
+    /// # Panics
+    /// Panics if `ValueType` is not `ValueType::String`.
+    #[allow(clippy::wildcard_enum_match_arm)] // Allow wildcard because there should be only one enum branch matches one specific type.
+    #[must_use]
+    pub fn into_string(self) -> String {
+        match self {
+            ValueType::String(string) => string,
+            _ => panic!("expect ValueType::String but get {self:?}"),
+        }
+    }
+
+    /// Turn the `ValueType` into `Raw`
+    /// # Panics
+    /// Panics if `ValueType` is not `ValueType::Raw`.
+    #[allow(clippy::wildcard_enum_match_arm)] // Allow wildcard because there should be only one enum branch matches one specific type.
+    #[must_use]
+    pub fn into_raw(self) -> Vec<u8> {
```

Comment:
We'd better to implement into_DetailType rather than do the deserialize work out side.

Reply:
You can submit an issue to document this problem.